### ISSUE TITLE
De-emphasizing Posts

### ISSUE THREAD

**Issue by ChaseWiseman** [royboy789/helium#15]
I love what I'm seeing of this project so far. Very exciting!

I think this would be a great opportunity to eliminate Posts as the central object/content type of WordPress. As most of us know, a lot of core components like pages, nav menus, attachments, etc... are just implementations of the original post component, which makes sense given WordPress' blog platform origins.

I feel that going forward it would make sense to stick with a more general _Content Type_ or _Content Component_ model, with _Posts_, _Pages_, and _Media_ being optional component implementations.

**Comment by jacobsantos** [comment]
:+1:

As I said with the WP Customizer component. You shouldn't have specialized types that duplicate members across types. You should have a type that has the common members or an implementation that allows arbitrary values that can be used in specialized types.

Meaning, you have a Content type that Pages, Attachments, etc where it makes sense and use the implementation in that Content type to fulfill the contract of the type requirements.

The idea should be to allow growth of types, while using less code and reusing as many components as possible.

**Comment by royboy789** [comment]
I agree fully, I think there should be no default content type at all, although media is going to be separated into its own component.

**Comment by mpeshev** [comment]
+1 for renaming Posts to something more generic and reasonable.

Also, platforms like Drupal try to get as many things as possible to be a Node, which is something that needs to be discussed separately. Right now a number of things make sense to be custom content types, but some would be better handled with custom tables, which is a problem to be discussed in #8

**Comment by markoheijnen** [comment]
Unsure if we even should implement post types. We should implement something that post types and taxonomies could inherit from.

**Comment by royboy789** [comment]
@markoheijnen I think WP by default right now assumes at least 1 Post Type.. I think there shouldn't be any.. or maybe that is a separate module of some sort "default content types" but the core module that we are building should not auto generate a custom post type.

**Comment by markoheijnen** [comment]
@royboy789 I agree on that. Post types is a CMS concept and not a framework  one. By that I don't it should be here. Something like a node from Drupal could maybe make sense.
